metrics/resource_view.go

```diff
@@ -57,8 +58,23 @@ type meters struct {
 // optionForResource, which must lock allMeters, but only needs to lock
 // resourceViews if a new meter needs to be created.
 var resourceViews = storedViews{}
-var allMeters = meters{
-	meters: map[string]*meterExporter{"": &defaultMeter},
+var allMeters = meters{}
+
+// Sizing is largely arbitrary - selected to easily cover a realistic number of revisions
+const cacheSize int = 4096
+
+func init() {
+	// lru.New() only throws on negative size
+	// The default meter is required - make sure it's not evicted
+	cache, _ := lru.NewWithEvict(cacheSize, func(key interface{}, value interface{}) {
+		if key == "" {
+			allMeters.lock.Lock()
+			defer allMeters.lock.Unlock()
+			allMeters.meters.Add(key, value)
```

Comment:
the `cache` is thread safe, so 1) don't need to lock 2) just use the `cache` var.

Reply:
Done.